fix tscircuit.com/search

Ayushjhawar8 opened the pull request:
/claim #901
fix #901

seveibar commented:
idk about this appearance, the query parameter "q" should be used in the url so searches can be linked to CC @MustafaMulla29

Ayushjhawar8 commented:
@seveibar about the appearance, does making it left align makes it better?
![image](https://github.com/user-attachments/assets/84110994-20cd-4965-8952-71785d99ea76)

seveibar commented:
@Ayushjhawar8 no, look at the github search etc. its not a dropdown, they used the page and allow scrolling etc

Ayushjhawar8 commented:
![image](https://github.com/user-attachments/assets/d2c226ef-8f1a-414d-a418-110b56947d73)

Ayushjhawar8 commented:
seve now can you check its deployed at https://tscircuit-nae2r0svr-tscircuit.vercel.app/search

seveibar commented:
When there is no search results because theyre loading, display a small circular loader in the space where search components are appearing, make sure its centered with padding

MustafaMulla29 commented:
@Ayushjhawar8 Hey, are you working on this?

Ayushjhawar8 commented:
@MustafaMulla29 yes

Ayushjhawar8 commented:
https://github.com/user-attachments/assets/2c517a0a-6002-4e30-afcc-c09b037ac60c
@seveibar please check

MustafaMulla29 reviewed (changes requested):
Maybe use the snippet card used in the trending and user-profile page and also use a better loader

seveibar commented:
@Ayushjhawar8 loader needs more vertical padding. The footer should not be jumping around and visible to the user, you can use min height etc but make sure the loader is visible

there are issues with the display of the component. @MustafaMulla29 is probably right that you should use an existing one otherwise we’ll be working on these details for a bit (eg Created text wraps below)

![image](https://github.com/user-attachments/assets/983617cc-43e7-4372-9c1d-0821dae3e571)

seveibar commented:
Yea the author displaying twice is sloppy, make sure to check your designs

MustafaMulla29 reviewed (changes requested):
'q'(query) should be visible in the url

Ayushjhawar8 commented:
https://github.com/user-attachments/assets/d05fba5f-a0cc-40e3-807a-eaa7c23814da

i tried to address all the comments. Let me know if any more changes needed.

MustafaMulla29 reviewed (commented):
Also the cards are too wider, you can take reference from the trending and user-profile pages. @seveibar what do you think?

Ayushjhawar8 commented:
seve i removed the loader and changed to skeleton used in trending page etc, also made the q search work better while sharing, also changed layout to 3 by 3

seveibar commented on src/components/SearchComponent.tsx:
```diff
@@ -91,17 +93,17 @@ const SearchComponent: React.FC<SearchComponentProps> = ({
 
   return (
-    <form onSubmit={handleSearch} className="relative">
+    <form onSubmit={handleSearch} className="relative w-full">
```
does this still need to be changed? Make sure to not change files that aren't related to the PR

Ayushjhawar8 replied:
oh yes this was when i was testing initially, let me remove this

seveibar commented on src/components/PageSearchComponent.tsx:
```diff
@@ -0,0 +1,133 @@
+    }
+  }, [searchQuery, location, setLocation])
+
+  useEffect(() => {
+    const params = new URLSearchParams(window.location.search)
+    const urlQuery = params.get("q")
+    if (urlQuery !== null && urlQuery !== searchQuery) {
+      setSearchQuery(urlQuery)
+    }
+  }, [location])
```
antipattern. You can just use `useState(params.get("q") ?? "")`

Remember, every useEffect triggers a render and makes the page slow

```suggestion
```

seveibar commented on src/components/PageSearchComponent.tsx:
```diff
@@ -0,0 +1,133 @@
+    if (newUrl !== location) {
+      setLocation(newUrl)
+    }
+  }, [searchQuery, location, setLocation])
```
hmm i don't like this approach, you're removing all other url parameters, why not use the URLSearchParams object to do a proper construction rather than direct string manipulation

Note: remove location and setLocation from the useEffect to avoid potential infinite loops. Keep useEffects simple

Ayushjhawar8 replied:
okay i used `URLSearchParams` to properly manage query parameters hope so.

seveibar commented on src/pages/search.tsx:
```diff
@@ -1,25 +1,57 @@
+                  Search Packages
+                </h1>
+              </div>
+              <p className="text-lg text-gray-600 mb-4">
+                Find circuit packages for your next project. Browse through our
+                collection of community-created circuit designs, footprints, and
+                components.
+              </p>
```
```suggestion
```

seveibar commented on src/pages/search.tsx:
```diff
@@ -1,25 +1,57 @@
+                components.
+              </p>
+              <div className="flex flex-wrap gap-3">
+                <Badge variant="secondary" className="px-3 py-1">
+                  <Filter className="w-3.5 h-3.5 mr-1" />
+                  <span>Search by Name</span>
+                </Badge>
```
```suggestion
```

3rd time i've mentioned removing this button that doesn't do anything. It looks like a button and doesn't do anything

Ayushjhawar8 replied:
its like the tag we have in trending page, i took inspiration from, well we can remove it if not want

seveibar commented on src/components/PageSearchComponent.tsx:
```diff
@@ -0,0 +1,133 @@
+    }
+  }
+
+  const shouldOpenInEditor = location === "/editor" || location === "/ai"
```
```suggestion
```

Ayushjhawar8 commented:
Seve please check: https://tscircuit-psthdvpw7-tscircuit.vercel.app/search

MustafaMulla29 reviewed (changes requested):
![Screenshot_2025-04-22-08-58-36-02_40deb401b9ffe8e1df2f1cc5ba480b12](https://github.com/user-attachments/assets/17ac39f2-b962-41f3-a5f7-6969dec8cd22)
'Search' text should be fitting in mobile screens, currently its wrapping below.
Also i think its better to remove the white background card of the full search and results showing part. Coz it aligns with the trending page. @seveibar what do you think?

seveibar commented:
Makes sense to be 👍 @MustafaMulla29

Ayushjhawar8 commented:
The search title for each package are overlapping due to `snippetCard` design, and its occuring everywhere its used.

I removed the white bg for consistency

Ayushjhawar8 commented on src/components/PageSearchComponent.tsx:
```diff
@@ -0,0 +1,156 @@
+    ["snippetSearch", searchQuery],
+    async () => {
+      if (!searchQuery) return []
+      const { data } = await axios.get("/snippets/search", {
+        params: { q: searchQuery },
+      })
```
i think this change will cause issue as this would double-append the base URL since it's already configured in the axios instance.

MustafaMulla29 reviewed (changes requested):
https://github.com/user-attachments/assets/9abefa0f-cb4f-4763-bb43-23bdb2bbcbb1

The 'q' query has some issues, not updating properly

seveibar commented:
Note the search box background is not white on mobile, can be fixed on followup
